scripts/validate_proconnect_allowlist.py

```
@@ -0,0 +1,278 @@
+        return None, None
+    text = comment.strip()
+    if text.lower().startswith("source:"):
+        text = text.split(":", 1)[1].strip()
+    # Tolerate any spacing around the ``|`` separator (``a|b``, ``a | b``, ...).
+    parts = re.split(r"\s*\|\s*", text, maxsplit=1)
+    if len(parts) == 2:
+        return parts[0].strip() or None, parts[1].strip() or None
+    return text or None, None
+
+
+def _source_suffix(comment):
```

Ruff (0.16.0)

[warning] Missing return type annotation for private function `_source_suffix`

Add return type annotation: `str`

(ANN202)